src/metabase/notification/payload/core.clj

```diff
@@ -133,18 +134,20 @@
    :admin_email          (public-settings/admin-email)
    :style                {:button (button-style (public-settings/application-color))}})
 
-(defmulti payload
+(defmulti notification-payload
   "Given a notification info, return the notification payload."
   {:arglists '([notification-info])}
   :payload_type)
 
-(mu/defn notification-payload :- ::NotificationPayload
-  "Realize notification-info with :context and :payload."
-  [notification :- ::Notification]
-  (assoc (select-keys notification [:payload_type :condition])
-         :creator (t2/select-one [:model/User :id :first_name :last_name :email] (:creator_id notification))
-         :payload (payload notification)
-         :context (default-context)))
+(defmulti transform-payload
+  "Transform the payload for a specific channel to consume."
+  {:arglists '([channel-type notification-payload])}
+  (fn [channel-type notification-payload]
+    [channel-type (:payload_type notification-payload)]))
```

Comment:
Taking the example of "the result of an action", this dispatch is too primitive.

My suggestion is to rather have a `payload-type` multi-method that this uses.

Reply:
Similarly I think that `channel-type` is too course as well, and what we really need is a `template-type` related to particular cases, like "v2 of the payload exposed to table data updated slack notification templates", say `:template/table-update-v2`.

This way both `notification-payload-type` and `template-payload-type` can have fully specified schemas.

Reply:
yea I like that idea about having payload type multimethod. I'm also thinking about whether having a hierachy will be nice. 

For example: `table-update-v2` can be a type, but if some channel wants to customize it, we can have `table-update-v2-slack` that is a child of `table-update-v2` and can transform the output of its parent to what it wants (we'll have to use methodical for this).

This will help us reuse code some what.